readthedocs/doc_builder/python_environments.py

```diff
@@ -72,6 +74,112 @@ def delete_existing_venv_dir(self):
             )
             shutil.rmtree(venv_dir)
 
+    def install_build_tools(self):
+        if settings.RTD_DOCKER_COMPOSE:
+            # Create a symlink for ``root`` user to use the same ``.asdf``
+            # installation than ``docs`` user. Required for local building
+            # since everything is run as ``root`` when using Local Development
+            # instance
+            cmd = [
+                'ln',
+                '-s',
+                os.path.join(settings.RTD_DOCKER_WORKDIR, '.asdf'),
+                '/root/.asdf',
+            ]
+            self.build_env.run(
+                *cmd,
+            )
+
+        for tool, version in self.config.build.tools.items():
+            version = version.full_version  # e.g. 3.9 -> 3.9.7
```

Comment:
```suggestion
        for tool, version_data in self.config.build.tools.items():
            version = version_data.full_version  # e.g. 3.9 -> 3.9.7
```

Seems weird to overwrite the `version` objects from the config, and makes the code harder to read. I'd use a new variable in the iterator or this:

```suggestion
        for tool, version in self.config.build.tools.items():
            full_version = version.full_version  # e.g. 3.9 -> 3.9.7
```